packages/net/writebuffer.pony

```diff
@@ -0,0 +1,69 @@
+use "collections"
+
+class WriteBuffer
+  """
+  A simple constant sized byte buffer to push Numbers into.
+  The data container is iso in order to keep it sendable(?)
+  Potential upgrades:
+  1. Have a list of data buffers of preset size
+  2. Arguably, seperate functions for _be and _le
+  3. A better way to get data out of the buffer, without passing a new array?
+  """
+  var data: Array[U8] iso
+  var _offset: USize
+  var _little_endian: Bool
+
+  new create(size: USize, little_endian: Bool = false) =>
+    """
+    Allocate the buffer
+    """
+    data = recover Array[U8].undefined(size) end
+    _offset = 0
+    _little_endian = little_endian
+
+  fun ref get_data(x: Array[U8] iso): Array[U8] iso^ =>
```

Comment:
@SeanTAllen - not quite - it's doing a destructive read, so it's setting the new value of `data` and returning the old value (or at least I believe that's the intent).

For destructive reads, I like the pattern of calling methods like this "take data" instead of "get data" since it implies that the caller will take ownership.

Also, it seems like this could be simplified by getting rid of the parameter and always using a new, zero-length array that is created inside the method.  Or is there a use case where it needs to be a custom array?